test/test_jit_fuser_te.py

```diff
@@ -1308,27 +1293,21 @@ def foo(x, mask):
             torch.bool,
         ]
         sizes = [(2,), (4, 4)]
-        for self_dtype, mask_dtype, device, size in product(dtypes, dtypes, self.devices, sizes):
-            try:
-                input_v = self.data_for(self_dtype, device, size=size)
-                val = self.data_for(val_dtype, device, size=size)
-                mask = self.data_for(torch.bool, device, size=size)
+        for self_dtype, device, scalar_val, size in product(dtypes, self.devices, [0.4, 3], sizes):
+            input_v = self.data_for(self_dtype, device, size=size)
+            mask = self.data_for(torch.bool, device, size=size)
 
-                def fn(input_v, val, mask):
-                    return torch.masked_fill(input_v, mask, val)
-                ref = fn(input_v, val, mask)
-            except Exception:
```

Comment:
I'm not sure I follow how this would hide a bug in the fuser implementation of the op. If it doesn't work in eager mode, what difference does it make for us? The fix might still be valid, but I don't understand why we need to remove this try-catch. Should not we have enough combinations where eager mode doesn't throw?

Reply:
it's because my code to test if it works in eager used the wrong name for a variable, so it would always throw

Reply:
I still not quite get it - why don't we then fix the variable name and keep the `try-except` block? I mean, it's ok to omit it If there is no combination of parameters with which the eager version throws, but I suspect the reasoning was slightly off and I'd like to clarify that part for myself.

Reply:
eager never throws, so i didnt think there was a reason to keep the try-except block,

Reply:
Sounds good then!